comp/core/configsync/configsyncimpl/sync_integration_test.go

```diff
@@ -89,7 +90,7 @@ func TestRunWithChan(t *testing.T) {
 
 func TestRunWithInterval(t *testing.T) {
 	// Legitimate use for NewConfig case where we want to have 2 independent config object mimicing 2 process communicating
-	configCore := pkgconfigmodel.NewConfig("test", "DD", strings.NewReplacer(".", "_")) // nolint: forbidigo
+	configCore := viperconfig.NewConfig("test", "DD", strings.NewReplacer(".", "_")) // nolint: forbidigo
```

Comment:
We're currently aiming at completly abstract viper from the config interface. We're working on a replacement, `nodetreemodel`, that can be use by simply setting a configuration setting. We should not have to edit the code base in order to switch from one to the other.

Reply:
@hush-hush I tried to remove the use `viperconfig.NewConfig` for this particular test as you wrote is a `Legitimate use for NewConfig`. We can not use a mock, because the mock avoid creating multiple mocks: 

```go
if isConfigMocked {
		// The configuration is already mocked.
		return &mockConfig{setup.Datadog()}
	}
```
https://github.com/DataDog/datadog-agent/blob/425b9da6574f88997e819480a1434051807dbd1c/pkg/config/mock/mock.go#L34-L40

The config component used during the test would reuse the mock created